internal/blockchain/ethereum/private_transaction_manager.go

```diff
@@ -0,0 +1,262 @@
+				"serverAddress": "http://$(hostname -i):%s",
+				"sslConfig": {
+					"tls": "OFF"
+			},
+				"communicationType" : "REST"
+			}
+		],
+		"peer": [
+			%s
+		],
+		"keys": {
+		"passwords": [],
+		"keyData": [
+				{
+					"privateKeyPath": "${DDIR}/keystore/tm.key",
+					"publicKeyPath": "${DDIR}/keystore/tm.pub"
+				}
+			]
+		},
+		"alwaysSendTo": [],
+		"bootstrapNode": false,
+		"features": {
+			"enableRemoteKeyValidation": false,
+			"enablePrivacyEnhancements": true
+		}
+	}
+EOF
+/tessera/bin/tessera -configfile ${DDIR}/tessera-config-09.json
+`, TmTpPort, TmQ2tPort, TmP2pPort, peerList)
+	filename := filepath.Join(outputDirectory, entrypoint)
+	file, err := os.OpenFile(filename, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	_, err = file.WriteString(content)
+	if err != nil {
+		return err
+	}
+	CopyTesseraEntrypointToVolume(ctx, outputDirectory, volumeName)
+	return nil
+}
+
+func CopyTesseraEntrypointToVolume(ctx context.Context, tesseraEntrypointDirectory, volumeName string) error {
+	if err := docker.MkdirInVolume(ctx, volumeName, ""); err != nil {
+		return err
+	}
+	if err := docker.CopyFileToVolume(ctx, volumeName, filepath.Join(tesseraEntrypointDirectory, entrypoint), ""); err != nil {
+		return err
+	}
+	return nil
+}
+
+func CreateQuorumEntrypoint(ctx context.Context, outputDirectory, volumeName, memberIndex string, chainId int) error {
+	discoveryCmd := "BOOTNODE_CMD=\"\""
+	if memberIndex != "0" {
+		discoveryCmd = fmt.Sprintf(`bootnode=$(curl http://geth_0:%s -s --connect-timeout 5 --max-time 10 --retry 5 --retry-connrefused --retry-delay 0 --retry-max-time 60 --fail --header "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method": "admin_nodeInfo", "params": [], "id": 1}' | grep -o "enode://[a-z0-9@.:]*")
+BOOTNODE_CMD="--bootnodes $bootnode"
+BOOTNODE_CMD=${BOOTNODE_CMD/127.0.0.1/geth_0}`, GethPort)
+	}
+
+	content := fmt.Sprintf(`#!/bin/sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+GOQUORUM_CONS_ALGO=$(echo "${GOQUORUM_CONS_ALGO:-clique}" | tr '[:lower:]')
+if [ "istanbul" == "$GOQUORUM_CONS_ALGO" ];
+then
+    echo "Using istanbul for consensus algorithm..."
+    export CONSENSUS_ARGS="--istanbul.blockperiod 5 --mine --miner.threads 1 --miner.gasprice 0 --emitcheckpoints"
+    export QUORUM_API="istanbul"
+elif [ "qbft" == "$GOQUORUM_CONS_ALGO" ];
+then
+    echo "Using qbft for consensus algorithm..."
+    export CONSENSUS_ARGS="--mine --miner.threads 1 --miner.gasprice 0 --emitcheckpoints"
+    export QUORUM_API="istanbul"
+elif [ "raft" == "$GOQUORUM_CONS_ALGO" ];
+then
+    echo "Using raft for consensus algorithm..."
+    export CONSENSUS_ARGS="--raft --raftblocktime 300 --raftport 53000"
+    export QUORUM_API="raft"
+elif [ "clique" == "$GOQUORUM_CONS_ALGO" ];
+then
+	echo "Using clique for consensus algorithm..."
+	export CONSENSUS_ARGS=""
+	export QUORUM_API="clique"
+fi
+
+TESSERA_URL=http://member%stessera
```

Comment:
This seems to imply Tessera would always be enabled if I chose Quorum as my ethereum blockchain protocol runtime rather than Besu.

I think that is not an appropriate assertion.

Choosing Besu or Quorum as your EVM is one choice.

Choosing whether you have an off-chain private transaction manager on top feels like a separate choice.
I think it's really important to keep those clearly distinguished.

Reply:
Agree that Quorum / Besu can run without a private transaction manager. We struggled slightly on this decision because we would strongly recommend that the better alternative is to run Geth / Besu clients instead of using Quorum if the choice is to run without a private transaction manager.

But agree on your point those choices should not be forced onto the user, will refactor and allow Quorum to start without Tessera.